### ISSUE TITLE
No subclass for minor

### ISSUE THREAD

**Issue by boldtrn** [openmaptiles/openmaptiles#413]
I have seen that there is no subclass for minor roads, like living_street, residential, and unclassified.

Especially living_street is often rendered differently. Also I like to apply different rules to residential and unclassified. Is there a reason that they are missing? Could we add them?

#82 seems to suggest that they existed some time ago?

**Comment by jirik** [comment]
Yes, `subclass` existed: https://github.com/openmaptiles/openmaptiles/commit/a109ffd72e22456d5d2bd5a3afeefc2b49e86eb9

**Comment by jirik** [comment]
Yes, `subclass` existed: https://github.com/openmaptiles/openmaptiles/commit/a109ffd72e22456d5d2bd5a3afeefc2b49e86eb9

I am not sure why @lukasmartinelli removed it, maybe because of size or [grouping](https://github.com/openmaptiles/openmaptiles/issues/388#issuecomment-351145551) issues.

We've partially recreated `subclass` in https://github.com/openmaptiles/openmaptiles/commit/137667b235fd7c8961491bdeb842c9e982419396 https://github.com/openmaptiles/openmaptiles/commit/1f129c418490e58f1c3324ce922bf5f436649879.

If you're interested in this, please start with making PR for living_street subclass only.

**Comment by boldtrn** [comment]
> maybe because of size 

I guess compared to 50 name tags in different languages one subclass should not be that big ;).

(this is probably worth it's own issue) BTW: The tile size of OMT is actually a bit concerning (it's bigger than many other vector tile providers). I did not debug this in greater detail but I expect due to providing many languages the size grew. Maybe it would be worthwhile to remove duplicated names. I have seen many example where the same name occurs multiple times in different languages. Which seems a bit like wasted space to me, I guess most renderers should be able to fall back to name if name:lang is not available and only name should be provided if name:lang is equal to name.

> If you're interested in this, please start with making PR for living_street subclass only.

I might have a look at this in the future, but I currently miss the time to contribute in more open source projects :).

**Comment by jirik** [comment]
We care about size a lot. Actually the 50 name tags did not mean big increase of size, because not every feature has name in 50 languages and also because vector-tile handles duplicate names effectively: https://github.com/mapbox/vector-tile-spec/tree/master/2.1#45-example
